testsuite/smoke-test/src/zkid.rs

```diff
@@ -460,66 +162,54 @@ async fn test_zkid_groth16_signature_transaction_submission_proof_signature_chec
-            "14933419822301565783764657928814181728459886670248956535955133596731082875810",
-        ],
-        [
-            "16616167200367085072660100259194052934821478809307596510515652443339946625933",
-            "1103855954970567341442645156173756328940907403537523212700521414512165362008",
-        ],
-    )
-    .unwrap();
-    let c = G1Bytes::new_unchecked(
-        "296457556259014920933232985275282694032456344171046224944953719399946325676",
-        "10314488872240559867545387237625153841351761679810222583912967187658678987385",
-    )
-    .unwrap();
-    let proof = Groth16Zkp::new(a, b, c);
-
-    let jwt_header = "eyJhbGciOiJSUzI1NiIsImtpZCI6InRlc3RfandrIiwidHlwIjoiSldUIn0".to_string();
-
-    let config = Configuration::new_for_devnet_and_testing();
-    let zk_sig = ZkIdSignature {
-        sig: ZkpOrOpenIdSig::Groth16Zkp(SignedGroth16Zkp {
-            proof: proof.clone(),
-            non_malleability_signature: ephemeral_signature.clone(), // Wrong signature
-            extra_field: "\"family_name\":\"Straka\",".to_string(),
-            exp_horizon_secs: config.max_exp_horizon_secs,
-            override_aud_val: None,
-            training_wheels_signature: Some(EphemeralSignature::ed25519(
-                tw_sk.sign(&proof).unwrap(),
-            )),
-        }),
-        jwt_header,
-        exp_timestamp_secs: 1900255944,
-        ephemeral_pubkey: ephemeral_public_key,
-        ephemeral_signature,
-    };
+    let esk = get_sample_esk();
+    zkid_sig.ephemeral_signature = EphemeralSignature::ed25519(esk.sign(&raw_txn).unwrap());
+
+    // Compute the training wheels signature if not present
+    match &mut zkid_sig.sig {
+        ZkpOrOpenIdSig::Groth16Zkp(proof) => {
+            proof.training_wheels_signature = Some(EphemeralSignature::ed25519(
+                tw_sk.sign(&proof.proof).unwrap(),
+            ));
+        },
+        ZkpOrOpenIdSig::OpenIdSig(_) => {},
+    }
 
-    let signed_txn = SignedTransaction::new_zkid(raw_txn, sender_zkid_public_key, zk_sig);
+    SignedTransaction::new_zkid(raw_txn, zkid_pk, zkid_sig)
+}
 
-    info!("Submit zero knowledge transaction");
-    info.client()
-        .submit_without_serializing_response(&signed_txn)
-        .await
-        .unwrap_err();
+async fn get_zkid_transaction(
+    get_pk_and_sig_func: fn() -> (ZkIdSignature, ZkIdPublicKey),
+) -> (ZkIdSignature, ZkIdPublicKey, LocalSwarm, SignedTransaction) {
+    let (tw_sk, mut swarm) = setup_local_net().await;
+
+    let (zkid_sig, zkid_pk) = get_pk_and_sig_func();
+
+    let mut info = swarm.aptos_public_info();
+    let signed_txn =
+        sign_zkid_transaction(&mut info, zkid_sig.clone(), zkid_pk.clone(), &tw_sk).await;
+
+    (zkid_sig, zkid_pk, swarm, signed_txn)
 }
 
-async fn test_setup(swarm: &mut LocalSwarm, cli: &mut CliTestFramework) -> Ed25519PrivateKey {
+async fn setup_local_net() -> (Ed25519PrivateKey, LocalSwarm) {
+    let (mut swarm, mut cli, _faucet) = SwarmBuilder::new_local(4)
```

Comment:
nit: don't really need 4 nodes?

Reply:
Well, once we enable JWK consensus, we might want to test that with 4 nodes here, no?

Reply:
JWK consensus is enabled for these tests.  Like the swarm will have Google's JWK's.

We don't use those JWKs as we use the JWK that we patch in.

Reply:
it's a nit so feel free to not change it :) 4 nodes just cost more $ in CI (and not providing extra coverage at the moment, since it uses a JWK patch)

Reply:
Okay. Will change to 1.